src/js/users/components/CreateUser.tsx

```diff
@@ -0,0 +1,54 @@
+import { DialogPortal, DialogTrigger } from "@radix-ui/react-dialog";
+import React from "react";
+import { useMutation } from "react-query";
+import { Dialog, DialogContent, DialogOverlay, DialogTitle, Icon } from "../../base";
+import { StyledButton } from "../../base/styled/StyledButton";
+import { create } from "../api";
+import { CreateUserForm } from "./CreateUserForm";
+
+type NewUserProps = {
+    /** The user's handle or username */
+    handle: string;
+    /** The user's password */
+    password: string;
+    /** Whether the user will be forced to reset their password on next login */
+    forceReset: boolean;
+};
+
+/**
+ * A dialog for creating a new user
+ */
+export default function CreateUser() {
+    const mutation = useMutation(create, {
+        onSuccess: () => {
+            history.replaceState({ state: !history.state.state }, "");
+        },
+    });
+
+    function handleSubmit({ handle, password, forceReset }: NewUserProps) {
+        mutation.mutate({ handle, password, forceReset });
+    }
+
+    function handleChange() {
+        mutation.reset();
+        history.replaceState({ state: !history.state?.state }, "");
```

Comment:
```suggestion
        history.replaceState({ createUser: !history.state?.state }, "");
```

I would give the variable in state are more meaningful name than state. Also consider if you want the button needs the ability to close the dialogue or not.